src/wallet/wallet.cpp

```diff
@@ -2459,9 +2461,9 @@ void CWallet::GetAvailableP2CSCoins(std::vector<COutput>& vCoins) const {
 /**
  * Test if the transaction is spendable.
  */
-static bool CheckTXAvailabilityInternal(const CWalletTx* pcoin, bool fOnlyConfirmed, int& nDepth)
+static bool CheckTXAvailabilityInternal(const CWalletTx* pcoin, bool fOnlySafe, int& nDepth)
 {
-    if (fOnlyConfirmed && !pcoin->IsTrusted()) return false;
+    if (fOnlySafe && !pcoin->IsTrusted()) return false;
```

Comment:
We could add a return param for the `IsTrusted` result like we are doing with the depth. So the function isn't called twice to set the safe field of the returned output.

Reply:
Yeah. `IsTrusted` should be extremely fast for the majority of txes (locktime=0), but have added it anyway.